clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -1130,8 +1130,96 @@ struct BitTest {
 
   static BitTest decodeBitTestBuiltin(unsigned BuiltinID);
 };
+
+// Returns the first convergence entry/loop/anchor instruction found in |BB|.
+// std::nullopt otherwise.
+std::optional<llvm::IntrinsicInst *> getConvergenceToken(llvm::BasicBlock *BB) {
```

Comment:
Don't see the point of wrapping this in optional instead of just returning null. It's no safer and now requires more syntax

Reply:
No strong opinion. My rational is if you read "optional", you know it's not always there. But I'm fine with the pointer, changed to pointer.